lib/devices/gadgets/fitness_equipment.dart

```diff
@@ -413,8 +413,8 @@ class FitnessEquipment extends DeviceBase {
         "ensuring that manufacturer name ($manufacturerName) contains manufacturer prefix ${descriptor!.manufacturerPrefix}",
       );
     }
-    return manufacturerName!.toLowerCase().contains(descriptor!.manufacturerPrefix.toLowerCase()) ||
-        descriptor!.manufacturerPrefix == "Unknown";
+    return manufacturerName?.toLowerCase().contains(descriptor!.manufacturerPrefix.toLowerCase()) ??
+        false || descriptor!.manufacturerPrefix == "Unknown";
```

Comment:
Due to operator precedence (https://support.wahoofitness.com/hc/en-us/articles/204280494-Android-Bluetooth-Troubleshooting) I think (not 100% sure, but kind of confident) we want a parentheses around `(manufacturerName?.toLowerCase().contains(descriptor!.manufacturerPrefix.toLowerCase()) ?? false)` otherwise the `??` is evaluated after the `||` so the way it is now it'll be `(false || descriptor!.manufacturerPrefix == "Unknown")`. (BTW to not be hypocritical: after I merge your PR I'll also make the `descriptor!` part robust, but I don't want to drag this PR out with that.)